### PR TITLE
statistics: not put pseudo table of partition table into cache

### LINKED ISSUE #44308: Many empty partitions cause statistics.PseudoTable to take up much memory
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Create 20 partitioned tables without data, each table has 60 columns and 2000 partitions.

### 2. What did you expect to see? (Required)

Since there is no data and statistics, no major change in memory usage.

### 3. What did you see instead (Required)

TiDB memory usage increases abnormally and continues to increase after each truncate of these tables. Most of memory are taken up by statistics.PseudoTable.

img_v2_63c2495b-5363-4cc1-84cd-49cc11cb813g

image

### 4. What is your TiDB version? (Required)

v6.5.2

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #44308

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/handle/handle.go [hunk 1]
```diff
@@ -1005,12 +1005,18 @@ func (h *Handle) GetPartitionStats(tblInfo *model.TableInfo, pid int64, opts ...
 	if !ok {
 		tbl = statistics.PseudoTable(tblInfo)
 		tbl.PhysicalID = pid
-		h.updateStatsCache(statsCache.update([]*statistics.Table{tbl}, nil, statsCache.version))
+		if tblInfo.GetPartitionInfo() == nil || h.statsCacheLen() < 64 {
```
Comment (hawkingrei) [inline statistics/handle/handle.go:1008]: ```h.statsCacheLen() < 64``` can avoid too much test case failing.

**Review by xuyifangreeneyes — APPROVED** [review]

File: statistics/handle/handle.go (hunk 1 above)
Comment (xuyifangreeneyes) [reply statistics/handle/handle.go:1008]: Maybe we can add some comments here.

**Review by time-and-fate — APPROVED** [review]

**Pull request merged**